model/client4.go

```diff
@@ -4355,6 +4355,24 @@ func (c *Client4) GetFileInfosForPost(postId string, etag string) ([]*FileInfo,
 	return list, BuildResponse(r), nil
 }
 
+// GetFileInfosForPost gets all the file info objects attached to a post, including deleted
+func (c *Client4) GetFileInfosForPostIncludeDeleted(postId string, etag string) ([]*FileInfo, *Response, error) {
+	r, err := c.DoAPIGet(c.postRoute(postId)+"/files/info"+"?include_deleted="+c.boolString(true), etag)
+	if err != nil {
+		return nil, BuildResponse(r), err
+	}
+	defer closeBody(r)
+
+	var list []*FileInfo
+	if r.StatusCode == http.StatusNotModified {
+		return list, BuildResponse(r), nil
+	}
+	if jsonErr := json.NewDecoder(r.Body).Decode(&list); jsonErr != nil {
+		return nil, nil, NewAppError("GetFileInfosForPostIncludeDeleted", "api.unmarshal_error", nil, jsonErr.Error(), http.StatusInternalServerError)
+	}
+	return list, BuildResponse(r), nil
+}
+
```

Comment:
Note, decided to implement completely seperate client function for the specific case for including deleted as this was the pattern that I noticed throughout the file ([ex](https://github.com/mattermost/mattermost-server/blob/54a88fb53268874aad436c77c72586ac24eb3e70/model/client4.go#L1919-L1950)), as well as the fact that changing the current function would deprecate current users of the library.

That being said, not sure that this is the best pattern for scalability down the line, especially if more options are anticipated.